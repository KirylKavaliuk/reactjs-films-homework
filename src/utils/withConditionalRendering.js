import React, { Component } from 'react';

export default function withConditionalRendering(WrappedComponent, keyFields) {
  return class extends Component {
    render() {
      if (keyFields.every(field => this.props[field] !== undefined)) {
        return <WrappedComponent { ...this.props }/>;
      }

      return null;
    }
  };
}
