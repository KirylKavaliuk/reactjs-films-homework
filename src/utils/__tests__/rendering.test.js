import React from 'react';
import renderer from 'react-test-renderer';

import withConditionalRendering from 'utils/rendering';

const Test = ({ array, string }) => (
  <div>
    { array }
    { string }
  </div>
);

it('renders description close shallow correctly', () => {
  const TestComponentArray = withConditionalRendering(Test, 'array');
  const TestComponentString = withConditionalRendering(Test, 'string');

  const arrayTree = renderer
    .create(<TestComponentArray array={[1, 2, 3]}/>)
    .toJSON();

  const stringTree = renderer
    .create(<TestComponentString string='str'/>)
    .toJSON();

  const nullTree = renderer
    .create(<TestComponentArray/>)
    .toJSON();

  expect(arrayTree).toMatchSnapshot();
  expect(stringTree).toMatchSnapshot();
  expect(nullTree).toMatchSnapshot();
});
