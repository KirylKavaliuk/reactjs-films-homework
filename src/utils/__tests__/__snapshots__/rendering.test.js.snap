// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders description close shallow correctly 1`] = `
<div>
  1
  2
  3
</div>
`;

exports[`renders description close shallow correctly 2`] = `
<div>
  str
</div>
`;

exports[`renders description close shallow correctly 3`] = `null`;
